### DISCUSSION TITLE
Configuring Giscus

### DISCUSSION THREAD

**Discussion by niz33** [cotes2020/jekyll-theme-chirpy#1189] [Q&A] [answered]
I'm using GitHub actions to deploy the site but can't seem to get Giscus to show up. I filled in the "repo", "repo_id", "category", and "category_id" variables in _config.yml and also manually created a discussion matching the pathname of a post. However, Giscus simply does not show up on the deployed site. I wonder if I'm doing something wrong or missing some steps.

**Comment by borosbence** [comment]
Hi!

I had updated my chirpy theme from 5.3 to 6.1.0 with git merge successfully, but my giscus comments not showing.
When I open the console I got the following error message (in Microsoft Edge):

> Refused to frame 'https://giscus.app/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'self'".

## Logs/Screenshots

Giscus frame in my website:

![Képernyőkép 2023-08-24 151714](https://github.com/cotes2020/jekyll-theme-chirpy/assets/39780892/ec4d7619-52d8-46c3-8ee6-86cf8944efe6)

Console error message:

![Képernyőkép 2023-08-24 151740](https://github.com/cotes2020/jekyll-theme-chirpy/assets/39780892/a3d268eb-8891-4477-845a-36702a4b65e3)

HTTP response:

![Képernyőkép 2023-08-24 151832](https://github.com/cotes2020/jekyll-theme-chirpy/assets/39780892/65adcebe-d001-43c0-ba1c-dafe05032ad8)

I searched for solutions, but it showed me some meta tags with HTTP headers, I don't know how to implement it, or this will fix it...
- https://stackoverflow.com/questions/42781949/refused-to-display-in-a-frame-because-an-ancestor-violates-the-following-content
- https://support.bigcommerce.com/s/question/0D54O000072mLgsSAE/embedded-checkout-error-refused-to-frame-because-an-ancestor-violates-the-following-content-security-policy-directive
- https://www.beautifulsoup.dev/blog/security/frame-ancestors-'self-%EC%98%A4%EB%A5%98'

FireFox show me the console message and doesn't load the giscus frame.

## Environment

| Command                           | Version |
|-----------------------------------|---------|
| `ruby -v`                         |3.3.2|
| `gem -v`                          |3.4.10 |
| `bundle -v`                       |2.4.10|
| `bundle exec jekyll -v`           |4.3.2|
| `bundle info jekyll-theme-chirpy` |6.1.0|


### Desktop

- OS: Windows
- Browser: Microsoft Edge116.0.1938.54

**Reply by borosbence** [reply]
I reinstalled Giscus on my repository and add brackets to the option values, and suddenly it works! ╰(*°▽°*)╯

**Comment by ottohellwig** [comment, marked as answer]
Hi @niz33! I took a look at your repository and browsed your config.yaml file, try:

- Removing the "" / '' from the giscus section, your repo, repo ID, category and category ID are all wrapped in quotes.
- Wrap giscus in active: giscus so that it looks like active: "giscus".
- Ensure giscus has permissions and is linked to your GitHub repository.

Hope these changes fix it for you!

**Reply by niz33** [reply]
Hi, thanks for the suggestions. I played around with the settings a bit and Giscus is working. It seems that double quotes in most places don't affect Giscus, but fiddling around with it solved the issue :)

**Reply by ottohellwig** [reply]
Awesome to hear! I hope you enjoy it, have a good week mate.
